src/api/routers/participantsRouter.ts

```diff
@@ -272,6 +280,55 @@ export function createParticipantsRouter() {
     }
   );
 
+  participantsRouter.get(
+    '/:participantId/apiRoles',
+    async (req: ParticipantRequest, res: Response) => {
+      const { participant } = req;
+      if (!participant?.siteId) {
+        return res.status(400).send('Site id is not set');
+      }
+
+      const apiRoles: ApiRoleDTO[] = await getApiRoles(participant);
+
+      return res.status(200).json(apiRoles);
+    }
+  );
+
+  const apiKeyCreateInputParser = z.object({ name: z.string(), roles: z.array(z.string()) });
+
+  participantsRouter.post(
+    '/:participantId/apiKeys/create',
+    async (req: ParticipantRequest, res: Response) => {
+      const { participant } = req;
+      if (!participant?.siteId) {
+        return res.status(400).send('Site id is not set');
+      }
+
+      const { name: keyName, roles: apiRoles } = apiKeyCreateInputParser.parse(req.body);
+
+      const traceId = getTraceId(req);
+      const currentUser = await findUserByEmail(req.auth?.payload?.email as string);
+      const auditTrail = await insertManageApiKeyAuditTrail(
+        participant,
+        currentUser!.id,
+        currentUser!.email,
+        AuditAction.Add,
+        keyName,
+        apiRoles,
+        traceId
+      );
+
+      if (!checkApiRoles(apiRoles, participant)) {
```

Comment:
Should the audit log be start before or after this check?

Reply:
hmm what do you think? maybe we only want to audit stuff that worked? perhaps worth raising with the team

Reply:
I would verge on having it before. That way we have a record if the api is being called incorrectly or maliciously.

Reply:
seems reasonable